MVCapplication/src/main/webapp/WEB-INF/resources/js/payment.js

```diff
@@ -0,0 +1,73 @@
+
+var $form = $('#payment-form');
+$form.find('.subscribe').on('click', payClicked);
+
+function payClicked(e) {
+    e.preventDefault();
+    /* Visual feedback */
+    $form.find('.subscribe').html(validating + ' <i class="fa fa-spinner fa-pulse"></i>').prop('disabled', true);
+    setTimeout(function(){$form.find('.subscribe').html(paymentSuccessful + ' <i class="fa fa-check"></i>'); window.location.href='/profile/getPremium';}, 500);
+
+}
+
+$('input[name=cardNumber]').payment('formatCardNumber');
+$('input[name=cardCVC]').payment('formatCardCVC');
+$('input[name=cardExpiry]').payment('formatCardExpiry');
+
+jQuery.validator.addMethod("cardNumber", function(value, element) {
+    return this.optional(element) || $.payment.validateCardNumber(value);
+}, invalidCardNumber);
+
+jQuery.validator.addMethod("cardExpiry", function(value, element) {
+    value = $.payment.cardExpiryVal(value);
+    return this.optional(element) || $.payment.validateCardExpiry(value.month, value.year);
+}, invalidExpireDate);
+
+jQuery.validator.addMethod("cardCVC", function(value, element) {
+    return this.optional(element) || $.payment.validateCardCVC(value);
+}, invalidCvc);
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') "invalidCvc" does not exist. Change its name or declare it so that its usage doesn't result in a "ReferenceError". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=javascript%3AS3827)